### ISSUE TITLE
Allow JsonParser configuration in JsonUtils

### ISSUE THREAD

**Issue by HansBrende** [jsonld-java/jsonld-java#237]
My use-case: I need JSON parsing to be lenient; i.e., I need to enable or disable the following features on the underlying `JsonFactory` in `JsonUtils`:

```java
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_BACKSLASH_ESCAPING_ANY_CHARACTER);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_COMMENTS);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_MISSING_VALUES);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_NUMERIC_LEADING_ZEROS);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_SINGLE_QUOTES);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_TRAILING_COMMA);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES);
JSON_FACTORY.enable(JsonParser.Feature.ALLOW_YAML_COMMENTS);
JSON_FACTORY.enable(JsonParser.Feature.IGNORE_UNDEFINED);
JSON_FACTORY.enable(JsonParser.Feature.INCLUDE_SOURCE_IN_LOCATION);
JSON_FACTORY.disable(JsonParser.Feature.STRICT_DUPLICATE_DETECTION);
```

Currently in order to do this, I need to use reflection to access the underlying `JsonFactory`. It would be nice to, instead, implement at least one of the following strategies in `JsonUtils`:

```java
public static void setLenient(boolean);
```
OR
```java
public static void enable(JsonParser.Feature);
public static void disable(JsonParser.Feature);
```
OR
```java
public static JsonFactory getJsonFactory();
```
OR
Add additional processing options for lenient parsing in `JsonLdOptions` (in which case the `org.eclipse.rdf4j.rio.jsonld.JSONLDParser` might require some tweaks as well).

Note: I am not accessing jsonld-java directly, or this would be an easier task; rather, I am using RDF4J which in turn uses jsonld-java through the `org.eclipse.rdf4j.rio.jsonld.JSONLDParser`. So I get stacktraces which look like this:

```
org.eclipse.rdf4j.rio.RDFParseException: Could not parse JSONLD
	at org.eclipse.rdf4j.rio.jsonld.JSONLDParser.parse(JSONLDParser.java:77)
	at org.apache.any23.extractor.rdf.BaseRDFExtractor.run(BaseRDFExtractor.java:196)
	... 36 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Illegal unquoted character ((CTRL-CHAR, code 10)): has to be escaped using backslash to be included in string value
 at [Source: (BufferedReader); line: 1, column: 147]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1804)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:663)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._throwUnquotedSpace(ParserMinimalBase.java:627)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._finishString2(ReaderBasedJsonParser.java:2045)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._finishString(ReaderBasedJsonParser.java:2016)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.getText(ReaderBasedJsonParser.java:278)
	at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer$Vanilla.deserialize(UntypedObjectDeserializer.java:672)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:527)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:364)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:29)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:3972)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2264)
	at com.fasterxml.jackson.core.JsonParser.readValueAs(JsonParser.java:1729)
	at com.github.jsonldjava.utils.JsonUtils.fromJsonParser(JsonUtils.java:196)
	at com.github.jsonldjava.utils.JsonUtils.fromReader(JsonUtils.java:173)
	at com.github.jsonldjava.utils.JsonUtils.fromInputStream(JsonUtils.java:154)
	at com.github.jsonldjava.utils.JsonUtils.fromInputStream(JsonUtils.java:111)
	at org.eclipse.rdf4j.rio.jsonld.JSONLDParser.parse(JSONLDParser.java:71)
	... 37 more
```

**Comment by ansell** [comment]
The best strategy in this case would be to create RDF4J configuration options and call from ``o.e.r.r.j.JSONLDParser.parse`` into ``JsonUtils.fromJsonParser`` instead, managing the JsonParser object lifecycle inside of RDF4J. I will look into submitting an RDF4J PR to do that.

**Comment by ansell** [comment]
A PR against Eclipse RDF4J targeted at its next minor version is open for review: https://github.com/eclipse/rdf4j/pull/1186

**Comment by fsteeg** [comment]
So https://github.com/eclipse/rdf4j/pull/1186 is merged and https://github.com/eclipse/rdf4j/commit/15bd63f1a41d2d5b2ec7e7f0315e17ece0b260f8 seems to be released since 2.5.0.

Is this issue resolved, @HansBrende?

**Comment by ansell** [comment]
Closing this as fixed, rdf4j is now using the `JsonUtils.fromJsonParser` method which enables it to control all of the parser configuration values that Jackson exposes.
